puzzle_generator/bytes_utils.py

```diff
@@ -5,14 +5,13 @@
 
 
 def int_to_bytes(in_value: int) -> bytes:
-    assert in_value >= 0  # nosec B101
+    if in_value < 0:
+        raise ValueError("in_value must be a positve integer")
     number_of_bytes = in_value.bit_length() // 8
     if 8 * number_of_bytes < in_value.bit_length():
         number_of_bytes += 1
-    assert number_of_bytes <= 256  # nosec B101
-    assert (
-        number_of_bytes - 1 <= in_value.bit_length() // 8 <= number_of_bytes
-    )  # nosec B101
+    if number_of_bytes > 256:
+        raise ValueError("in_value must be 256 bytes or less")
```

Comment:
```suggestion
    if number_of_bytes > 255:
        raise ValueError("in_value must be 255 bytes or less")
```
My bad: the limit should be `255` (256 does not fit into one byte).

As above: what do you think about adding such test to [`test_bytes_utils.py`](https://github.com/vil02/puzzle_generator/blob/0353bba4f79b5780099e2135f77eacd0841f2cca/tests/test_bytes_utils.py):
```python3
def test_int_to_bytes_raises_when_input_has_more_than_256_bytes():
    with pytest.raises(ValueError, match="in_value must be 255 bytes or less"):
        bu.int_to_bytes(2 ** (8 * 255))
```
`2 ** (8 * 255)` is the _smallest_ integer, which does not fit into 255 bytes.